source/slang/syntax.cpp

```diff
@@ -1579,7 +1579,11 @@ void Type::accept(IValVisitor* visitor, void* extra)
                 }
```

Comment:
See my comments on #251 for how I think the actual substitution logic can be improved. Not necessary to implement as part of this change, but I want to go on record as saying there is a Right Answer here (because I tried to plan for it in the design of `DeclaredSubtypeWitness`).